core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -351,10 +352,7 @@ default SafeCloseable push(boolean runCallbacks) {
 
         // Put the oldCtx back before throwing an exception.
         RequestContextThreadLocal.set(oldCtx);
-        throw new IllegalStateException(
-                "Trying to call object wrapped with context " + this + ", but context is currently " +
-                "set to " + oldCtx + ". This means the callback was called from " +
-                "unexpected thread or forgetting to close previous context.");
+        return throwIllegalContextPushing(this, oldCtx);
```

Comment:
Ah here's where I first saw the pattern - see my comment here https://github.com/line/armeria/pull/2371#discussion_r363069843

Using `return` instead of `throw` doesn't seem to improve verbosity but makes the code much less readable.

Reply:
Yes, it's now much readable. Thanks!